compiler/record-minmax/src/RecordMinMax.cpp

```diff
@@ -400,7 +493,67 @@ void RecordMinMax::profileData(const std::string &mode, const std::string &input
     throw std::runtime_error("HDF5 error occurred.");
   }
 
-  update_quantparam(_observer.get(), mode, min_percentile, max_percentile);
+  update_quantparam(getObserver(), mode, min_percentile, max_percentile);
+}
+
+void RecordMinMax::profileDataInParallel(const std::string &mode,
+                                         const std::string &input_data_path, float min_percentile,
+                                         float max_percentile)
+{
+  assert(_interpreters.size() == _threads_size);
+  assert(_observers.size() == _threads_size);
+  const auto vector_input_data = importH5Data(input_data_path);
+  const auto num_records = vector_input_data.size();
+  const auto input_nodes = loco::input_nodes(_module->graph());
+
+  // Start parallel part
+  std::cout << _threads_size << " concurrent threads are supported.\n";
+
+  if (num_records < _threads_size)
+  {
+    _threads_size = num_records;
+  }
+  uint32_t records_batch = static_cast<int>(num_records / _threads_size);
+
+  std::vector<std::thread> threads;
+  for (uint32_t t = 0; t < _threads_size; ++t)
+  {
+    if (t < _threads_size - 1)
+    {
+      threads.emplace_back(interpret_batch_records, records_batch * t, records_batch * (t + 1),
+                           _interpreters[t].get(), vector_input_data, input_nodes);
+    }
+    else
+    {
+      threads.emplace_back(interpret_batch_records, records_batch * t, num_records,
+                           _interpreters[t].get(), vector_input_data, input_nodes);
+    }
+  }
+
+  for (int i = 0; i < _threads_size; ++i)
+    threads.at(i).join();
```

Comment:
Multi-threading part.

I guess that `interpret_batch_records` can be defined in this function to reduce parameters. For example,

```cpp
auto interpret_batch_records = [&vector_input_data, &input_nodes, &batch_size] (Interpreter *interp, uint32_t batch_index) {
...
};

```

Reply:
Done